stellar/txobserver/main.go

```diff
@@ -23,51 +25,66 @@ type Observer struct {
 	store  *store.DB
 	log    *slog.Entry
 
-	bridgeAccountCreateSequence uint32
+	ledgerSequence uint32
+	catchup        bool
 }
 
 func NewObserver(
 	ctx context.Context,
 	bridgeAccount string,
-	bridgeAccountCreateSequence uint32,
 	client *horizonclient.Client,
 	store *store.DB,
 ) *Observer {
 	o := &Observer{
-		ctx:                         ctx,
-		bridgeAccount:               bridgeAccount,
-		bridgeAccountCreateSequence: bridgeAccountCreateSequence,
-		client:                      client,
-		store:                       store,
-		log:                         slog.DefaultLogger.WithField("service", "stellar_txobserver"),
+		ctx:           ctx,
+		bridgeAccount: bridgeAccount,
+		client:        client,
+		store:         store,
+		log:           slog.DefaultLogger.WithField("service", "stellar_txobserver"),
+	}
+
+	ledgerSeq, err := o.store.GetLastLedgerSequence(context.Background())
+	if err != nil {
+		o.log.Fatalf("Unable to load last ledger sequence from db: %v", err)
+	}
+
+	if ledgerSeq == 0 {
+		// Perform catchup on the first call to ProcessNewLedgers
+		o.catchup = true
+	} else {
+		o.ledgerSequence = ledgerSeq
 	}
 
 	return o
 }
 
 func (o *Observer) ProcessNewLedgers() {
 	for o.ctx.Err() == nil {
-		if ledgerSeq, err := o.store.GetLastLedgerSequence(context.Background()); err != nil {
-			o.log.Errorf("Unable to load last ledger sequence from db: %v", err)
-		} else {
-			if ledgerSeq == 0 {
-				ledgerSeq = o.bridgeAccountCreateSequence
+		if o.catchup {
+			err := o.catchupLedgers()
+			if err != nil {
+				o.log.WithFields(slog.F{"error": err}).Error("Error catching up")
```

Comment:
What is the strategy if catchup fails?
If `ProcessNewLedgers` fails we should treat that as a fatal error and stop the program because it can create bad edge case, what do you think?

Reply:
If catchup fails it will continue running the loop until the `ProcessNewLedgers` is successfully done. Basically `ProcessNewLedgers` will block until it's fully synced to the latest ledger available in connected Horizon.